umbraco/Umbraco-CMS
# Ensure package view path is generated with consideration the current section

This is a partial fix for #13951, resolving a client-side bug where we didn't consider the current section when generating the URL for a plugin tree. That was causing issues when two packages included trees with the same alias.

I've added an additional parameter to `treeService.getTreePackageFolder` to provide the current section alias, which is then used when finding the tree data in the server vars collection. While it's possible that these methods are being used in third-party code, I think it's highly unlikely, which is why I've added the new param first rather than last, so that the order of params makes logical sense - section, tree, then method. 

I've also added an additional property to the PluginTree model, so that we pass the section alias on the server vars collection. This is required because a plugin alias and section name are independent values so we can't compare these to find the package folder (eg "MyPackage" and "myPackageAlias" are perfectly valid values, but one can't be used to determine the other). Section alias will match the route segment, so all good to compare against.

To test:
1. spin up a clean 10/11/12 site (let's go for 12)
2. add the below code 
3. build, run, login,
4. update the administrator group to grant access to SectionA and SectionB
5. verify that clicking the root menu node in SectionA attempts to load /App_Plugins/SectionB/backoffice/settings/edit.html

1. checkout these changes
2. repeat 2-4 above
3. verify that clicking the root menu node in Section A and B attempts to load the correct view (which verifies these changes work when trees are added in custom sections)
4. verify other sections work correctly (eg settings:logViewer)

```csharp
internal sealed class Composer : IComposer
{
    public void Compose(IUmbracoBuilder builder)
    {
        _ = builder.Sections().Append<SectionA>();
        _ = builder.Sections().Append<SectionB>();
    }
}

internal sealed class SectionA : ISection
{
    public string Alias => "section-a";
    public string Name => "SectionA";
}

internal sealed class SectionB : ISection
{
    public string Alias => "sectionbAlias";
    public string Name => "SectionB";
}

[PluginController("SectionB")]
[Tree("sectionbAlias", "settings", SortOrder = 5)]
public class SectionBSettingsTreeController : TreeController
{
    public SectionBSettingsTreeController(ILocalizedTextService localizedTextService, UmbracoApiControllerTypeCollection umbracoApiControllerTypeCollection, IEventAggregator eventAggregator) : base(localizedTextService, umbracoApiControllerTypeCollection, eventAggregator)
    {
    }
    public ActionResult<TreeNode?> CreateRootNode(FormCollection queryStrings, string section, string routePath, string icon, bool hasChildren = false, string? menuUrl = null)
    {
        ActionResult<TreeNode?>? root = CreateRootNode(queryStrings);

        if (root.Value is null)
        {
            return root;
        }

        root.Value.RoutePath = section + "/" + routePath + "/overview";
        root.Value.Icon = icon;
        root.Value.HasChildren = hasChildren;
        root.Value.MenuUrl = menuUrl;

        return root;
    }

    protected override ActionResult<TreeNodeCollection> GetTreeNodes(string id, [ModelBinder(typeof(HttpQueryStringModelBinder))] FormCollection queryStrings) => new TreeNodeCollection
    {
        CreateTreeNode("1000", id, queryStrings, "SectionB tree node", Constants.Icons.Folder, false, string.Empty),
    };
    protected override ActionResult<MenuItemCollection> GetMenuForNode(string id, [ModelBinder(typeof(HttpQueryStringModelBinder))] FormCollection queryStrings) => null!;
}

[PluginController("SectionA")]
[Tree("section-a", "settings", SortOrder = 60)] 
public sealed class SectionASettingsTreeController : TreeController
{
    public SectionASettingsTreeController(ILocalizedTextService localizedTextService, UmbracoApiControllerTypeCollection umbracoApiControllerTypeCollection, IEventAggregator eventAggregator) : base(localizedTextService, umbracoApiControllerTypeCollection, eventAggregator)
    {
    }
    public ActionResult<TreeNode?> CreateRootNode(FormCollection queryStrings, string section, string routePath, string icon, bool hasChildren = false, string? menuUrl = null)
    {
        ActionResult<TreeNode?>? root = CreateRootNode(queryStrings);

        if (root.Value is null)
        {
            return root;
        }

        root.Value.RoutePath = section + "/" + routePath + "/overview";
        root.Value.Icon = icon;
        root.Value.HasChildren = hasChildren;
        root.Value.MenuUrl = menuUrl;

        return root;
    }

    protected override ActionResult<TreeNodeCollection> GetTreeNodes(string id, [ModelBinder(typeof(HttpQueryStringModelBinder))] FormCollection queryStrings) => new TreeNodeCollection
    {
        CreateTreeNode("1000", id, queryStrings, "SectionA tree node", Constants.Icons.Folder, false, string.Empty),
    };
    protected override ActionResult<MenuItemCollection> GetMenuForNode(string id, [ModelBinder(typeof(HttpQueryStringModelBinder))] FormCollection queryStrings) => null!;
}

```
